### PR TITLE
community: Microsoft Azure Document Intelligence updates

### PR DESCRIPTION
**Pull request by ruoccofabrizio**
- **Description:** Update Azure Document Intelligence implementation by Microsoft team and RAG cookbook with Azure AI Search

### CODE REVIEW

**Comment by ruoccofabrizio** [comment]
@luzhang06  @toothache

File: cookbook/rag_semantic_chunking.ipynb
Comment (baskaryan) [inline cookbook/rag_semantic_chunking.ipynb:1]: could we add something to file name to make it clear it's using azure doc intelligence?

File: libs/community/langchain_community/document_loaders/parsers/doc_intelligence.py [hunk 1]
```diff
@@ -15,24 +18,40 @@ def __init__(
         api_endpoint: str,
         api_key: str,
         api_version: Optional[str] = None,
-        api_model: str = "prebuilt-layout",
+        model_id: str = "prebuilt-layout",
```
Comment (baskaryan) [inline libs/community/langchain_community/document_loaders/parsers/doc_intelligence.py:21]: this is breaking change, what's value of doing this?

**Review by baskaryan — COMMENTED** [review]
thanks @ruoccofabrizio!! few comments

File: cookbook/rag_semantic_chunking.ipynb
Comment (luzhang06) [reply cookbook/rag_semantic_chunking.ipynb:1]: Sure. Updated the notebook file name.

**Comment by luzhang06** [comment]
Hi @baskaryan, can you please help with this PR to move forward? We have addressed your previous comments. Let us know if you have any further comments.

File: libs/community/langchain_community/document_loaders/parsers/doc_intelligence.py (hunk 1 above)
Comment (toothache) [reply libs/community/langchain_community/document_loaders/parsers/doc_intelligence.py]: Reverted the API change for this PR.

**Comment by toothache** [comment]
@baskaryan , can you take another look at the change? I have reverted the API change and resolved the conflicts.

Let us know if there's any more concern. Thanks!

**Comment by eyurtsev** [comment]
Comandeering to lint

**Review by eyurtsev — APPROVED** [review]

File: libs/community/langchain_community/document_loaders/parsers/doc_intelligence.py [hunk 2]
```diff
@@ -17,22 +20,42 @@ def __init__(
         api_version: Optional[str] = None,
         api_model: str = "prebuilt-layout",
         mode: str = "markdown",
```
Comment (eyurtsev) [inline libs/community/langchain_community/document_loaders/parsers/doc_intelligence.py:22]: We should be using `*` as much as we can to make sure that we dont' break user code accidentally

```suggestion
        mode: str = "markdown",
        *,
```

**Review by baskaryan — APPROVED** [review]

**Comment by baskaryan** [comment]
some failing tests

**Comment by toothache** [comment]
> some failing tests

Thanks! The UT has been fixed.

**Comment by toothache** [comment]
Fixing lint.

**Comment by toothache** [comment]
Thanks so much for the review! @baskaryan  @eyurtsev

**Comment by eyurtsev** [comment]
Apologies for taking so long!

**Pull request merged**